src/shared/unit-file.c

```diff
@@ -21,3 +22,54 @@ bool unit_type_may_template(UnitType type) {
                       UNIT_TIMER,
                       UNIT_PATH);
 }
+
+int unit_validate_symlink(const char *filename, const char *target) {
```

Comment:
since this logs (which I think is unexpected for such a library-like function), maybe call this unit_validate_symlink_and_warn()?